shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java

```diff
@@ -0,0 +1,234 @@
+
+import alluxio.client.block.BlockMasterClient;
+import alluxio.client.block.options.ReportWorkerOptions;
+import alluxio.client.block.options.ReportWorkerOptions.ReportWorkerInfoField;
+import alluxio.client.block.options.ReportWorkerOptions.WorkerRange;
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.util.FormatUtils;
+import alluxio.wire.ReportWorkerInfo;
+
+import com.google.common.base.Strings;
+import org.apache.commons.cli.CommandLine;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Prints Alluxio capacity information.
+ */
+public class CapacityCommand {
+  private static final int INDENT_SIZE = 4;
+  private static final String USAGE = "alluxio fsadmin report capacity [filter arg]\n"
+      + "Report Alluxio capacity information.\n"
+      + "Where [filter arg] is an optional argument, if no arguments passed in, "
+      + "capacity information of all workers will be printed out.\n"
+      + "[filter arg] can be one of the following:\n"
+      + "    -live                         Live workers\n"
+      + "    -lost                         Lost workers\n"
+      + "    -worker <worker_ip_addresses> Specified workers, IP addresses separated by \",\"\n";
+
+  private BlockMasterClient mBlockMasterClient;
+  private PrintStream mPrintStream;
+  private int mIndentationLevel = 0;
+  private StringBuilder mStringBuilder;
+
+   /**
+   * Creates a new instance of {@link CapacityCommand}.
+   *
+   * @param blockMasterClient client to connect to block master
+   * @param printStream stream to print summary information to
+   */
+  public CapacityCommand(BlockMasterClient blockMasterClient, PrintStream printStream) {
+    mBlockMasterClient = blockMasterClient;
+    mPrintStream = printStream;
+    mStringBuilder = new StringBuilder();
+  }
+
+  /**
+   * Runs report capacity command.
+   *
+   * @param cl CommandLine to get client options
+   * @return 0 on success, 1 otherwise
+   */
+  public int run(CommandLine cl) throws IOException {
+    if (cl.hasOption("h")) {
+      System.out.println(USAGE);
+      return 0;
+    }
+    ReportWorkerOptions options = getOptions(cl);
+    printWorkerCapacityInfo(options);
+    return 0;
+  }
+
+  /**
+   * Prints Alluxio capacity information.
+   *
+   * @param options ReportWorkerOptions to define the report worker info range
+   */
+  public void printWorkerCapacityInfo(ReportWorkerOptions options) throws IOException {
+    List<ReportWorkerInfo> workerInfolist = mBlockMasterClient.getReportWorkerInfoList(options);
+
+    // Summarize the worker capacity information
+    long sumCapacityBytes = 0;
+    long sumUsedBytes = 0;
+    Map<String, Long> sumTotalBytesOnTiersMap = initializeTierMap();
+    Map<String, Long> sumUsedBytesOnTiersMap = initializeTierMap();
+
+    // Cache the worker capacity information
```

Comment:
what do you mean by "cache"?

Reply:
changed to "collect".
I hope to print aggregated info first and then print cached worker info, so I use cache here.